### PR TITLE
[Process] Fix stopping a process on Windows

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15617 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by stof** [comment]
Is the HHVM failure in the process component test related to the change you made here ?

**Comment by nicolas-grekas** [comment]
HHVM failure is unrelated, it would need an other fix

File: src/Symfony/Component/Process/Tests/AbstractProcessTest.php [hunk 1]
```diff
@@ -838,10 +860,10 @@ public function testMethodsThatNeedATerminatedProcess($method)
             $process->stop(0);
             $this->fail('A LogicException must have been thrown');
```
Comment (Tobion) [inline src/Symfony/Component/Process/Tests/AbstractProcessTest.php]: These fail conditions in the tests you changed are not useless pretty much. So they can be removed.

Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Tests/AbstractProcessTest.php]: not sure: `throw $e` still needs $e to be defined

File: src/Symfony/Component/Process/Process.php [hunk 2]
```diff
@@ -157,8 +157,16 @@ public function __construct($commandline, $cwd = null, array $env = null, $stdin
...
+        if ($this->isRunning()) {
+            $this->doSignal(15, false);
+            usleep(1000);
+        }
+        if ($this->isRunning()) {
+            $this->doSignal(9, false);
+            usleep(1000);
+        }
+
+        // Don't call ->stop() nor ->close() since it's blocking and we don't care about waiting for the subprocess here
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Process.php]: calling for review/comments on this change

File: src/Symfony/Component/Process/Tests/AbstractProcessTest.php (hunk 1 above)
Comment (Tobion) [inline src/Symfony/Component/Process/Tests/AbstractProcessTest.php]: I see. Those tests should be rewritten with try...finally in 3.0

**Comment by stof** [comment]
Thank you @nicolas-grekas.

**Pull request merged**